src/Workspaces/Core/Portable/FindSymbols/FindReferences/FindReferencesSearchEngine_DocumentProcessing.cs

```diff
@@ -31,7 +31,7 @@ internal partial class FindReferencesSearchEngine
                     var symbol = symbolAndFinder.symbolAndProjectId;
                     var finder = symbolAndFinder.finder;
 
-                    await ProcessDocumentAsync(document, symbol, finder).ConfigureAwait(false);
+                    await ProcessDocumentAsync(document, model, symbol, finder).ConfigureAwait(false);
```

Comment:
thank you @sharwell 

one question, is this only place where you give in model from document.GetSematnicModel? no other place?

if that is true.. there seems an issue on weak reference caching.

Reply:
I agree.  This indicates a much deeper and more concerning bug.  As we can see in this method, the model is acquired at the top, and then held all the way to the finally block.  It is part of the contract of Roslyn that as long as this guy is held around that Document.TryGetSemanticModel should succeed and must always return that same semantic model.

This approach is working around that not being working.  But having that contract be violable is a deep and concerning issue.

Reply:
@CyrusNajmabadi Agreed, something is up. If we want the perf optimization that's fine, but this is incredibly easy to break. :-)